compiler/rustc_codegen_ssa/src/mir/intrinsic.rs

```diff
@@ -331,6 +344,53 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 use crate::common::AtomicOrdering::*;
                 use crate::common::{AtomicRmwBinOp, SynchronizationScope};
 
+                let invalid_monomorphization = |ty| {
+                    bx.tcx().dcx().emit_err(InvalidMonomorphization::BasicIntegerType {
+                        span,
+                        name,
+                        ty,
+                    });
+                };
+
+                let parse_const_generic_ordering = |ord: ty::Value<'tcx>| {
+                    let discr = ord.valtree.unwrap_branch()[0].unwrap_leaf();
+                    let ord = discr.to_atomic_ordering();
+                    // We have to translate from the intrinsic ordering to the backend ordering.
+                    use rustc_middle::ty::AtomicOrdering;
```

Comment:
I wonder if `rustc_middle::ty::AtomicOrdering` and `rustc_codegen_ssa::common::AtomicOrdering` could be merged into a single enum. Doesn't have to be done in this PR.